Fix bug with AM and PM classes

jadebuckwalter opened the pull request:
Detect difference between AM and PM period names so they show up in the
correct places

jadebuckwalter committed:
Fix bug with AM and PM classes

Detect difference between AM and PM period names so they show up in the
correct places

psvenk reviewed (changes requested):
Thank you @jadebuckwalter for the changes. Instead of hard-coding "l" for AM/PM classes, I think it would be a better idea to detect regular periods (01, 02, 03, 04) and fall back to a strict equality check, similar to how it worked before 9d2d9a89af. Maybe we could do a regex search on `period` for `/0\d/` to detect a regular period, and if so, see if the same period number is a substring of the default name. Your thoughts?

jadebuckwalter commented:
@psvenk Sure, that makes sense.

psvenk committed:
get_period_name: use a more robust algorithm

Instead of relying solely on the last character of the period name,
include explicit support for multiple period name formats so as to
reduce the rate of false positives (e.g., "Before School" and "After
School" being detected as the same period).

psvenk reviewed (approved):
Looks good to me now. @jadebuckwalter, feel free to squash and merge (maybe using something like my commit message) once you have verified that this works for you.
